core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java

```diff
@@ -0,0 +1,160 @@
+package ca.jrvs.apps.grep;
+
+import org.apache.log4j.BasicConfigurator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.*;
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class JavaGrepImp implements JavaGrep {
+
+    final Logger logger = LoggerFactory.getLogger(JavaGrep.class);
+
+    protected String regex;
+    protected String rootPath;
+    protected String outFile;
+
+    public static void main(String[] args) {
+        if (args.length != 3) {
+            throw new IllegalArgumentException("USAGE: JavaGrep regex rootPath outFile");
+        }
+
+        // User default logger config
+        BasicConfigurator.configure();
+
+        JavaGrepImp javaGrepImp = new JavaGrepImp();
+        javaGrepImp.setRegex(args[0]);
+        javaGrepImp.setRootPath(args[1]);
+        javaGrepImp.setOutFile(args[2]);
+
+        try {
+            javaGrepImp.process();
+        } catch(Exception e) {
+            javaGrepImp.logger.error(e.getMessage(), e);
```

Comment:
javaGrepImp.logger.error(~e.getMessage()~ "meaningful message", e);

Reply:
Fixed it.